### DISCUSSION TITLE
When selecting a long paragraph, nvda could speak the first few words and last few words, instead of just saying number of words

### DISCUSSION THREAD

**Discussion by LittleStar-VIP** [nvaccess/nvda#14763] [Ideas]
### Is your feature request related to a problem? Please describe.
1. Say I found the following paragraph from the wikipedia

A screen reader is a form of assistive technology (AT)[1] that renders text and image content as speech or braille output. Screen readers are essential to people who are blind,[2] and are useful to people who are visually impaired,[2] illiterate, or have a learning disability.[3] Screen readers are software applications that attempt to convey what people with normal eyesight see on a display to their users via non-visual means, like text-to-speech,[4] sound icons,[5] or a braille device.[2] They do this by applying a wide variety of techniques that include, for example, interacting with dedicated accessibility APIs, using various operating system features (like inter-process communication and querying user interface properties), and employing hooking techniques.[6]

2. I want to select that paragraph by using ctrl+shift+down arrow
3. after selection, nvda only says 776 words selected

The problem is I would like a double confirmation of whether or not I have correctly selected the paragraph that I am intended to select

### Describe the solution you'd like
instead of just saying 776 words selected, nvda could say something like
'''
A screen reader is a form of assistive technology (AT)[1] that renders text and image content ……various operating system features (like inter-process communication and querying user interface properties), and employing hooking techniques.[6] 776 words

'''

### Describe alternatives you've considered
create an option in nvda that allow user to configure even selecting a long paragraph, nvda will still report everything when doing the selection

A further note, as the same behavior apply to when we press nvda+c to report what is in the clipboard
To avoid creating a similar issue here, I think the above suggestion also apply to nvda+c

**Comment by Adriani90** [comment]
I think it is not a user friendly way to report the beginning and the end of text. Note that in some texts for example the same sentence could appear multiple times, so it is not a guaratee that you selected the right text if you hear the last selected words.

Additionally, reporting the text of a very big selected paragraph causes performance issues and even crashes. That'ts why NVDA changed this behavior. The nvda+c command has a buffer of 1023 characters before NVDA begins to report the number or character instead of the text. For paragraph Selection this is truncated after 511 characters.

As an alternative you can use the Windows clipboard history which you can enable in Windows / system / clipboard. Then you can review what you copied to clipboard.

**Comment by LittleStar-VIP** [comment]
If user have to review the clipboard history in order to know what have been copied, this is also not user friendly at all isn't it?
So in that sense if nvda detected that the string length > 511, nvda just speak the first 511 characters?

**Comment by seanbudd** [comment]
This needs further discussion before accepting this as an issue.
